baseapp/baseapp.go

```diff
@@ -240,6 +242,8 @@ func (app *BaseApp) InitChain(req abci.RequestInitChain) (res abci.ResponseInitC
 	app.setDeliverState(abci.Header{ChainID: req.ChainId})
 	app.setCheckState(abci.Header{ChainID: req.ChainId})
 
+	app.consensusParams = req.ConsensusParams
```

Comment:
This should go away, and instead we should pass it into setDeliverState and setCheckState (assuming I'm right about what we should do, as per the above comments).  And we should clone ConsensusParams too before passing them.